src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -310,3 +314,16 @@ def __call__(self, *args, **kwargs):
             inputs=kwargs,
         )
         return result.output
+
+    def invoke(self, data: dict):
+        from promptflow._sdk._submitter.test_submitter import TestSubmitterViaProxy
+
+        # if args:
+        #     raise UserErrorException("Flow can only be called with keyword arguments.")
+
+        submitter = TestSubmitterViaProxy(flow=self, flow_context=self.context)
```

Comment:
we will need to be able to cache this

Reply:
may I ask what do you mean by cache. You mean cache like Flowinvoker?

Reply:
added cache

Reply:
offline synced with Xingzhi, remove executor cache in this pr